lib/auth/auth_with_roles.go

```diff
@@ -66,7 +68,10 @@ func (a *ServerWithRoles) action(namespace string, resource string, action strin
 // even if they are not admins, e.g. update their own passwords,
 // or generate certificates, otherwise it will require admin privileges
 func (a *ServerWithRoles) currentUserAction(username string) error {
-	if a.hasLocalUserRole(a.context.Checker) && username == a.context.User.GetName() {
+	if hasLocalUserRole(a.context.Checker) && username == a.context.Identity.GetIdentity().Username {
+		return nil
+	}
+	if hasRemoteUserRole(a.context.Checker) && username == a.context.UnmappedIdentity.GetIdentity().Username {
```

Comment:
What's the implication of this change? Wouldn't this mean that, for example, a remote user has permissions to change the local user's password as long as they have the same username? I may be missing something.

Reply:
Nice catch, didn't think about that.
It also turns out I was confusing "username" with "ssh login name", so I renamed that request field and it's no longer compared to the caller username.
Changes to `currentUserAction` reverted.